src/main/java/bio/terra/workspace/db/ControlledResourceDao.java

```diff
@@ -0,0 +1,83 @@
+package bio.terra.workspace.db;
+
+import bio.terra.workspace.service.controlledresource.exception.DuplicateControlledResourceException;
+import bio.terra.workspace.service.controlledresource.model.ControlledResourceMetadata;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ControlledResourceDao {
+  private final Logger logger = LoggerFactory.getLogger(ControlledResourceDao.class);
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public ControlledResourceDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public void createControlledResource(ControlledResourceMetadata controlledResource) {
+    final String sql =
+        "INSERT INTO workspace_resource "
+            + "(workspace_id, resource_id, associated_app, is_visible, owner, attributes) values "
+            + "(:workspaceId, :resourceId, :associatedApp, :isVisible, :owner, :attributes)";
+    final var params =
+        new MapSqlParameterSource()
+            .addValue("workspaceId", controlledResource.workspaceId())
+            .addValue("resourceId", controlledResource.resourceId())
+            .addValue("isVisible", controlledResource.isVisible())
+            .addValue("associatedApp", controlledResource.associatedApp().orElse(null))
+            .addValue("owner", controlledResource.owner().orElse(null))
+            .addValue("attributes", controlledResource.attributes().orElse(null));
```

Comment:
I suppose I need to build a map into JSON for this field. Not sure I need it yet though.